lightning-invoice/src/utils.rs

```diff
@@ -381,31 +399,43 @@ where
 /// channel with a higher or equal inbound capacity than `min_inbound_capacity_msat` exists
 /// * If any public channel exists, the returned `RouteHint`s will be empty, and the sender will
 /// need to find the path by looking at the public channels instead
-fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Option<u64>) -> Vec<RouteHint>{
+fn filter_channels<L: Deref>(
+	channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Option<u64>, logger: &L
+) -> Vec<RouteHint> where L::Target: Logger {
 	let mut filtered_channels: HashMap<PublicKey, &ChannelDetails> = HashMap::new();
 	let min_inbound_capacity = min_inbound_capacity_msat.unwrap_or(0);
 	let mut min_capacity_channel_exists = false;
 
+	log_debug!(logger, "Considering {} channels for invoice route hints", channels.len());
```

Comment:
Hmm I think it'd be pretty useful to also add some logging in `_create_phantom_invoice` for this line to not be too confusing when generating phantom invoices. Say you're generating a phantom invoice using 3 participating nodes with 3 channels each or something. Then this line will be repeated 3 times ("Considering 3 channels for invoice route hints") instead of saying that saying that 9 channels were considered which I think is what people would expect instead.

I think that will be pretty confusing with no context, if you don't also have a good understanding of how the code for phantom invoice generation is structured, and that the channels for each participated node is filtered separately.

I think that can become clearer if there's also some logging in `_create_phantom_invoice` that indicates that route hints will be generated for X participating nodes.
The context will also be clearer by also adding a line of logging in the `_create_phantom_invoice` `for loop` right before the `filter_channels` call, which indicates that channels for participating node with node id {} are about to be considered for invoice route hints.